File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
  const lowerAlpha = 'abcdefghijklmnopqrstuvwxyz';
  const upperAlpha = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'

  let result = '';
  [...s].forEach((ele) => {
    if(ele === ' ') result += ' ';
    else{
      if(ele === ele.toUpperCase()) result += upperAlpha[(upperAlpha.indexOf(ele) + n) % 26]
      else result += lowerAlpha[(lowerAlpha.indexOf(ele) + n) % 26]
    }})
  return result;
}

console.log(solution("z A Z", 2))
console.log(solution("a Z b c F", 2))

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    big_alpha = list('ABCDEFGHIJKLMNOPQRSTUVWXYZ')
    small_alpha = list('abcdefghijklmnopqrstuvwxyz')

    result = ''
    for ele in s:
        if ele == ' ':
            result += ' '
        else:
          result += big_alpha[(big_alpha.index(ele) + n) % 26] if ele.isupper() else small_alpha[(small_alpha.index(ele) + n) % 26]
      
    return result